spec/services/delivery_request_service_spec.rb

```diff
@@ -1,212 +1,147 @@
+    it "calls the provider to send an email" do
+      email_parameters = { address: email.address,
+                           subject: email.subject,
+                           body: email.body }
+      expect(default_provider).to receive(:call)
+                              .with(hash_including(email_parameters))
+                              .and_return(:sending)
+      described_class.call(email: email)
     end
-  end
 
-  describe "#provider" do
-    it "defaults to the PseudoProvider" do
-      expect(subject.provider).to eq(PseudoProvider)
+    it "can send an email to a configured provider" do
+      config = EmailAlertAPI.config.email_service.merge(provider: "notify")
+      expect(NotifyProvider).to receive(:call).and_return(:sending)
+      described_class.call(email: email, config: config)
     end
 
-    it "can be configured" do
-      subject = described_class.new(config: config.merge(provider: "NOTIFY"))
-      expect(subject.provider).to eq(NotifyProvider)
+    it "can prefix the subject with a configured prefix" do
+      config = EmailAlertAPI.config.email_service.merge(email_subject_prefix: "[TEST] ")
+      expect(default_provider).to receive(:call)
+                              .with(hash_including(subject: "[TEST] #{email.subject}"))
+                              .and_return(:sending)
+      described_class.call(email: email, config: config)
     end
-  end
 
-  describe "#subject_prefix" do
-    it "defaults to be an empty string" do
-      expect(subject.subject_prefix).to eq("")
+    it "can be configured to override a recepients email address" do
+      address = "override@example.com"
+      config = EmailAlertAPI.config.email_service.merge(email_address_override: address)
+      expect(default_provider).to receive(:call)
+                              .with(hash_including(address: address))
+                              .and_return(:sending)
+      described_class.call(email: email, config: config)
     end
 
-    it "can be configured" do
-      subject = described_class.new(config: config.merge(email_subject_prefix: "STAGING - "))
-      expect(subject.subject_prefix).to eq("STAGING - ")
-    end
-  end
+    it "returns true when the overrider can provide an email address" do
+      overrider_double = instance_double(
+        described_class::EmailAddressOverrider,
+        destination_address: "test@example.com",
+      )
+      allow(described_class::EmailAddressOverrider)
+        .to receive(:new)
+        .and_return(overrider_double)
 
-  shared_examples "records a statistic" do |status|
-    it "records a #{status} statistic" do
-      expect(GovukStatsd).to receive(:increment).with("delivery_attempt.status.#{status.underscore}")
-      subject.call(email: email)
+      expect(described_class.call(email: email)).to be(true)
     end
-  end
 
-  describe "#call" do
-    let!(:email) { create(:email) }
+    it "returns false when the overrider can't provide an email address" do
```

Comment:
```suggestion
    it "returns false when no email was sent due to filtering" do
```

It took a little effort to track down what the boolean is used for, so helpful to indicate this in the tests.

Reply:
Yeah, the boolean is a bit nasty. I've updated as per your suggestion.